src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1197 @@
+        hidden_states = torch.cat([text_hidden, image_hidden], dim=1)
+        return hidden_states
+
+
+class Beit3Embedder(nn.Module):
+    def __init__(self, modules, dim=1):
+        super().__init__()
+        self.dim = dim
+        self.first = modules[0]
+        self.second = modules[1]
+        self.split_position = -1
+
+    def forward(self, hidden_states, positions, multiway_split_position=None):
+        if multiway_split_position is not None:
+            self.split_position = multiway_split_position
+        if self.split_position == -1:
+            return self.first(hidden_states, positions=positions)
+        if self.split_position == 0:
+            return self.second(hidden_states, positions=positions)
+        text_hidden, image_hidden = torch.split(
+            hidden_states,
+            [self.split_position, hidden_states.size(self.dim) - self.split_position],
+            dim=self.dim,
+        )
+        y1, y2 = self.first(text_hidden, positions=positions), self.second(image_hidden, positions=positions)
+        return torch.cat([y1, y2], dim=self.dim)
+
+
+class Beit3VisionEmbedding(nn.Module):
+    """Image to Patch Embedding"""
+
+    def __init__(self, config):
+        super().__init__()
+        img_size = (config.img_size, config.img_size)
+        patch_size = (config.patch_size, config.patch_size)
+        num_patches = (img_size[1] // patch_size[1]) * (img_size[0] // patch_size[0])
+        self.patch_shape = (img_size[0] // patch_size[0], img_size[1] // patch_size[1])
+        self.img_size = img_size
+        self.patch_size = patch_size
+        self.num_patches = num_patches
+
+        self.proj = nn.Conv2d(config.in_chans, config.embed_dim, kernel_size=patch_size, stride=patch_size)
+        self.mask_token = nn.Parameter(torch.zeros(1, 1, config.embed_dim))
+        self.cls_token = nn.Parameter(torch.zeros(1, 1, config.embed_dim))
+
+    def num_position_embeddings(self):
+        if self.cls_token is None:
+            return self.num_patches
+        else:
+            return self.num_patches + 1
+
+    def forward(self, hidden_states: torch.Tensor, masked_position: bool = None) -> torch.Tensor:
+        hidden_states = self.proj(hidden_states).flatten(2).transpose(1, 2)
+
+        batch_size, seq_len, _ = hidden_states.size()
+
+        if masked_position is not None:
+            mask_token = self.mask_token.expand(batch_size, seq_len, -1)
+            w = masked_position.unsqueeze(-1).type_as(mask_token)
+            hidden_states = hidden_states * (1 - w) + mask_token * w
+
+        if self.cls_token is not None:
+            cls_tokens = self.cls_token.expand(batch_size, -1, -1)
+            hidden_states = torch.cat((cls_tokens, hidden_states), dim=1)
+
+        return hidden_states
+
+
+class Beit3PositionalEmbedding(nn.Embedding):
+    def forward(
+        self,
+        hidden_states: torch.Tensor,
+        positions: torch.Tensor = None,
+    ):
```

Comment:
nit: can go on one line 

```suggestion
    def forward(self, hidden_states: torch.Tensor, positions: torch.Tensor = None):
```